### PR TITLE
Factor out guide and acq candidates mask

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

This makes publicly-available versions of the guide and acq star candidates masks. These have previously been stuck in a class method that ended up getting copied and modified for application code.

## Interface impacts

Adds two new functions that are useful for external applications.

### Unit tests

- [x] Mac
```
(ska3) ➜  proseco git:(factor-out-candidates-mask) git rev-parse HEAD                                                        
c683dac86fb19b12350c333e2080a5d4763825b5
(ska3) ➜  proseco git:(factor-out-candidates-mask) pytest
======================================================== test session starts ========================================================
platform darwin -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /Users/aldcroft/git
configfile: pytest.ini
plugins: timeout-2.2.0, anyio-4.3.0
collected 167 items                                                                                                                 

proseco/tests/test_acq.py .....................................                                                               [ 22%]
proseco/tests/test_catalog.py ..........................................                                                      [ 47%]
proseco/tests/test_core.py ...........................                                                                        [ 63%]
proseco/tests/test_diff.py ......                                                                                             [ 67%]
proseco/tests/test_fid.py ...............                                                                                     [ 76%]
proseco/tests/test_guide.py ....................................                                                              [ 97%]
proseco/tests/test_mon_full_cat.py ....                                                                                       [100%]

======================================================= 167 passed in 21.69s ========================================================
```

Independent check of unit tests by Jean
- [x] Linux

```
ska3-jeanconn-fido> git rev-parse HEAD
aeee4352722a4077a828637573c608c06bf27689
ska3-jeanconn-fido> pytest
======================================= test session starts ========================================
platform linux -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /proj/sot/ska/jeanproj/git
configfile: pytest.ini
plugins: anyio-4.3.0, timeout-2.2.0
collected 167 items                                                                                             

proseco/tests/test_acq.py .....................................                                           [ 22%]
proseco/tests/test_catalog.py ..........................................                                  [ 47%]
proseco/tests/test_core.py ...........................                                                    [ 63%]
proseco/tests/test_diff.py ......                                                                         [ 67%]
proseco/tests/test_fid.py ...............                                                                 [ 76%]
proseco/tests/test_guide.py ....................................                                          [ 97%]
proseco/tests/test_mon_full_cat.py ....                                                                   [100%]

======================================== 167 passed in 129.81s (0:02:09)
```

### Functional tests

Functionally tested in https://github.com/sot/chandra_aca/pull/170. In particular, test the handling of star catalogs without `mag` and `mag_err` columns.

### CODE REVIEW

File: proseco/acq.py [hunk 1]
```diff
@@ -178,6 +183,41 @@ def boxes_overlap(
...
+    stars : Table
+        Table of stars
+
+    Returns
+    -------
+    ok : np.array of bool
+        Mask of acceptable stars
+    """
+    # Allow for stars from proseco StarsTable.from_stars() with `mag` and `mag_err`,
+    # or native agasc.get_agasc_cone() stars table.
```
Comment (jeanconn) [inline proseco/acq.py:202]: Is this tested?

File: proseco/guide.py [hunk 2]
```diff
@@ -97,6 +102,60 @@ def get_guide_catalog(obsid=0, **kwargs):
...
+    mag_err = (
+        stars["mag_err"] if "mag_err" in stars.colnames else stars["MAG_ACA_ERR"] / 100
+    )
+
+    ok = (
+        (stars["CLASS"] == 0)
+        & (mag > 5.2)
+        & (mag < faint_mag_limit)
+        & (mag_err < 1.0)
+        & (stars["ASPQ1"] < 20)  # Mag err < 1.0 mag
```
Comment (jeanconn) [inline proseco/guide.py:151]: I think the previous comments were wrong/shifted and that persists here.

File: proseco/acq.py (hunk 1 above)
Comment (taldcroft) [reply proseco/acq.py:202]: Yes. Proseco supplies `stars` with `mag` and `mag_err` columns while `chandra_aca` (https://github.com/sot/chandra_aca/pull/170) plotting uses native `agasc` names `MAG` and `MAG_ACA_ERR`.

File: proseco/guide.py (hunk 2 above)
Comment (taldcroft) [reply proseco/guide.py]: Fixed.

File: proseco/acq.py (hunk 1 above)
Comment (jeanconn) [reply proseco/acq.py:202]: So functionally tested vial the chandra_aca PR.

Comment (taldcroft) [reply proseco/acq.py:202]: Correct.  I updated the description.

**Pull request merged**